x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,384 @@
+    ({ title, color, iconType }) => (): void => {
+      dispatchToaster({
+        type: 'addToaster',
+        toast: {
+          id: uuid.v4(),
+          title,
+          color,
+          iconType,
+        },
+      });
+    },
+    [dispatchToaster]
+  );
+  const [loadingList, exceptionList, exceptionItems, fetchList] = useExceptionList({
+    http,
+    id: selectedListId,
+    namespaceType: selectedListNamespaceType,
+    filterOptions,
+    onError: onDispatchToaster({
+      color: 'danger',
+      title: i18n.FETCH_LIST_ERROR,
+      iconType: 'alert',
+    }),
+  });
+
+  const setSelectedListType = useCallback(
+    (type: ToggleId): void => {
+      dispatch({
+        type: 'updateSelectedListType',
+        listType: type,
+      });
+    },
+    [dispatch]
+  );
+
+  const setIsModalOpen = useCallback(
+    (isOpen: boolean): void => {
+      dispatch({
+        type: 'updateModalOpen',
+        isOpen,
+      });
+    },
+    [dispatch]
+  );
+
+  const setExceptionToDelete = useCallback(
+    (id: string | null): void => {
+      dispatch({
+        type: 'updateExceptionToDelete',
+        id,
+      });
+    },
+    [dispatch]
+  );
+
+  const onFetchList = useCallback(
+    ({ id, namespaceType }: ApiProps): void => {
+      if (fetchList != null && id != null && namespaceType != null) {
+        fetchList({
+          listId: id,
+          listNamespaceType: namespaceType,
+        });
+      }
+    },
+    [fetchList]
+  );
+
+  const onFiltersChange = useCallback(
+    (filter: { filter: string; tags: string[] }): void => {
+      dispatch({
+        type: 'updateFilterOptions',
+        filterOptions: filter,
+      });
+    },
+    [dispatch]
+  );
+
+  const onAddExceptionItem = useCallback((): void => {
+    setIsModalOpen(true);
+  }, [setIsModalOpen]);
 
+  const onEditExceptionItem = useCallback(
+    (exception: ExceptionListItemSchema): void => {
+      dispatch({
+        type: 'updateExceptionToEdit',
+        exception,
+      });
+
+      setIsModalOpen(true);
+    },
+    [setIsModalOpen]
+  );
+
+  const onCloseExceptionModal = useCallback(
+    ({ actionType, listId, listNamespaceType }): void => {
+      setIsModalOpen(false);
+
+      // TODO: This callback along with fetchList can probably get
+      // passed to the modal for it to call itself maybe
+      if (actionType === ModalAction.CREATE && listId != null && onAssociateList != null) {
+        onAssociateList(listId);
+      }
+
+      onFetchList({ id: listId, namespaceType: listNamespaceType });
+    },
+    [setIsModalOpen, onFetchList, onAssociateList]
+  );
+
+  const onDeleteException = useCallback(
+    async ({ id, namespaceType }: { id: string; namespaceType: NamespaceType }) => {
+      try {
+        const deleteTask = new AbortController();
+        setExceptionToDelete(id);
+        await deleteExceptionListItemById({ http, id, namespaceType, signal: deleteTask.signal });
+        setExceptionToDelete(null);
```

Comment:
@spong @yctercero Do we want to show a success toaster? I'm asking because we recently added a success toaster for closing alerts and I don't know if this is a pattern we also want to follow here.

Reply:
We could go a couple ways with this. TBH I'd probably just put the `Remove` button in a loading state and disable the `Edit` button, and then animate the exception item out on success and only show a toast in the failure case. Showing a success toast is kind of irrelevant since the user should see the UI disappear anyway.

Reply:
I like that suggestion @spong. I'll add that in.

Reply:
Updated!